Moq.AutoMock/Resolvers/MockResolver.cs

```diff
@@ -32,19 +34,21 @@ public MockResolver(MockBehavior mockBehavior, DefaultValue defaultValue, bool c
         /// <param name="context">The resolution context.</param>
         public void Resolve(MockResolutionContext context)
         {
-            if (context is null) throw new ArgumentNullException(nameof(context));
-
-            if (!(context.Value is null)) return;
```

Comment:
Do we not still need this?
```suggestion
            if (context.Value is not null || context.RequestType == typeof(string))
            {
                return;
            }
```

Reply:
With this change there is no longer a need/desire to check the Value property. These types of checks prevented a resolver from legitimately returning null for a service. Now it will stop at the first resolver that sets the Value property, regardless of what value is set.